packages/oauth/oauth_server.js

```diff
@@ -135,7 +135,34 @@ OAuth._checkRedirectUrlOrigin = redirectUrl => {
   );
 };
 
-const middleware = (req, res, next) => {
+// Parses request payload to use in the handler
+//
+// @param {Object} req incoming oauth request
+//
+const parseRequestData = req => new Promise((resolve, reject) => {
+  if (req.method === 'POST') {
+    let rawBody = '';
+    req
+      .on('data', (chunk) => {
+        rawBody += chunk;
+      })
+      .on('end', () => {
+        const json = rawBody.split('&').reduce((arr, kv) => {
```

Comment:
Can't we use node's built-in querystring parser?
https://nodejs.org/api/querystring.html#querystring_querystring_parse_str_sep_eq_options

Reply:
Thanks for taking a look.
Sure, I'll fix that.